src/openrct2/cmdline/BenchUpdate.cpp

```diff
@@ -0,0 +1,160 @@
+        {
+            state.SkipWithError("Failed to load file!");
+        }
+
+        std::vector<LogicTimings> timings(1);
+        timings.reserve(100);
+        int currentTimingIdx = 0;
+        for (auto _ : state)
+        {
+            if (timings[currentTimingIdx].CurrentIdx == (LOGIC_UPDATE_MEASURMENTS_COUNT - 1))
+            {
+                timings.resize(timings.size() + 1);
+                currentTimingIdx++;
+            }
+            LogicTimings* timingToUse = &timings[currentTimingIdx];
+            context->GetGameState()->UpdateLogic(timingToUse);
+        }
+        state.SetItemsProcessed(state.iterations());
+        auto accumulator = [timings](LogicTimePart part) -> double {
+            std::chrono::duration<double> timesum;
+            for (const auto& timing : timings)
+            {
+                for (const auto& PartTime : timing.TimingInfo.at(part))
+                    timesum += PartTime;
+            }
+            return std::chrono::duration_cast<std::chrono::milliseconds>(timesum).count();
+        };
+        state.counters["NetworkUpdateAcc_ms"] = accumulator(LogicTimePart::NetworkUpdate);
+        state.counters["DateAcc_ms"] = accumulator(LogicTimePart::Date);
+        state.counters["ScenarioAcc_ms"] = accumulator(LogicTimePart::Scenario);
+        state.counters["ClimateAcc_ms"] = accumulator(LogicTimePart::Climate);
+        state.counters["MapTilesAcc_ms"] = accumulator(LogicTimePart::MapTiles);
+        state.counters["MapStashProvisionalElementsAcc_ms"] = accumulator(LogicTimePart::MapStashProvisionalElements);
+        state.counters["MapPathWideFlagsAcc_ms"] = accumulator(LogicTimePart::MapPathWideFlags);
+        state.counters["PeepAcc_ms"] = accumulator(LogicTimePart::Peep);
+        state.counters["MapRestoreProvisionalElementsAcc_ms"] = accumulator(LogicTimePart::MapRestoreProvisionalElements);
+        state.counters["VehicleAcc_ms"] = accumulator(LogicTimePart::Vehicle);
+        state.counters["MiscAcc_ms"] = accumulator(LogicTimePart::Misc);
+        state.counters["RideAcc_ms"] = accumulator(LogicTimePart::Ride);
+        state.counters["ParkAcc_ms"] = accumulator(LogicTimePart::Park);
+        state.counters["ResearchAcc_ms"] = accumulator(LogicTimePart::Research);
+        state.counters["RideRatingsAcc_ms"] = accumulator(LogicTimePart::RideRatings);
+        state.counters["RideMeasurmentsAcc_ms"] = accumulator(LogicTimePart::RideMeasurments);
+        state.counters["NewsAcc_ms"] = accumulator(LogicTimePart::News);
+        state.counters["MapAnimationAcc_ms"] = accumulator(LogicTimePart::MapAnimation);
+        state.counters["SoundsAcc_ms"] = accumulator(LogicTimePart::Sounds);
+        state.counters["GameActionsAcc_ms"] = accumulator(LogicTimePart::GameActions);
+        state.counters["NetworkFlushAcc_ms"] = accumulator(LogicTimePart::NetworkFlush);
+        state.counters["ScriptsAcc_ms"] = accumulator(LogicTimePart::Scripts);
```

Comment:
Maybe we should have something that converts from an enum of `LogicTimePart` into the corresponding string, to be less error prone. That would also allows us to do a for on the enum updating all counters
```c++
for (<all_enums>)
    state.counters[ToLogicPartString(enum)] = accumulator(enum);
```